aflr/helper_classes.py

```diff
@@ -0,0 +1,70 @@
+from .api_request import APIRequest
+
+
+class ListableResource(APIRequest):
+    def __init__(self):
+        super().__init__()
+
+    @classmethod
+    def list(cls, **args):
+        if args:
+            return cls._get_request(request_params=args)
+        else:
+            return cls._get_request()
+
+
+class CreatableResource(APIRequest):
+    def __init__(self):
+        super().__init__()
+
+    @classmethod
+    def create(cls, **params):
+        return cls._post_request(json=params)
+
+
+class RetrievableResource(APIRequest):
+    def __init__(self):
+        super().__init__()
+
+    @classmethod
+    def retrieve(cls, scriptId, section=None, parameters=None):
+        params = parameters or {}
+        params.update({"scriptId": scriptId})
+
+        if section:
+            params.update({"section": section})
+
+        if hasattr(cls, "file_url"):
+            return cls._get_request(path_param=cls.file_url, request_params=params)
+        else:
+            return cls._get_request(request_params=params)
+
+
+class DownloadableResource(APIRequest):
+    def __init__(self):
+        super().__init__()
+
+    @classmethod
+    def download(cls, scriptId, section=None, parameters=None, destination="."):
+        parameters = parameters or {}
+
+        try:
+            audio_files = cls.retrieve(scriptId, section, parameters)
+            audio_files.keys()
+        except Exception:
+            raise TypeError(
+                "Error retriving the audiofiles. The response object is not a dict"
```

Comment:
Error retrieving the audio files. Make sure you can retrieve them with the same parameters and try again.

is a better message I think

Reply:
changed